### PR TITLE
add "piccolo user list" command

### PR DESCRIPTION
**Pull request by waldner**

### CODE REVIEW

File: piccolo/apps/user/commands/list.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from piccolo.apps.user.tables import BaseUser
+
+
+def list():
+    """
+    List existing users.
+    """
+    users = BaseUser.select(
+        BaseUser.all_columns(exclude=[BaseUser.password])
+    ).run_sync()
```
Comment (dantownsend) [inline piccolo/apps/user/commands/list.py:10]: I'd probably make this `async`.

File: piccolo/apps/user/commands/list.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    if len(users) < 1:
+        return
```
Comment (dantownsend) [inline piccolo/apps/user/commands/list.py:13]: Is it worth printing out something like 'No users'?

File: piccolo/apps/user/commands/list.py (hunk 1 above)
Comment (dantownsend) [inline piccolo/apps/user/commands/list.py:10]: We can sort the results - I guess `username` is most useful?

File: piccolo/apps/user/commands/list.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    for user in users:
+        for column in ref_order:
+            if len(str(user[column])) > width[column]:
+                width[column] = len(str(user[column]))
+
+    format_string = "|".join([f"{{:<{width[w]}}}" for w in ref_order])
+    print(format_string.format(*[w for w in ref_order]))
+    for user in users:
+        print(format_string.format(*[str(user[w]) for w in ref_order]))
```
Comment (dantownsend) [inline piccolo/apps/user/commands/list.py:26]: I think what you've done looks cool.

Screenshot 2024-01-22 at 15 34 00

What do you think about adding a space either side of the `|` symbols? Only if it doesn't add loads of complexity.

Comment (dantownsend) [reply piccolo/apps/user/commands/list.py:26]: There's a file called `piccolo/utils/printing.py` which has utilities for this kind of thing. We don't have anything which does exactly what you're doing, but maybe it's worth add it there so it's reusable?

**Review by dantownsend — COMMENTED** [review]
This looks great - thanks!

I left a few comments.

File: piccolo/apps/user/commands/list.py (hunk 1 above)
Comment (waldner) [reply piccolo/apps/user/commands/list.py:10]: I've checked other `user` subcommands and they are synchronous.

File: piccolo/apps/user/commands/list.py (hunk 2 above)
Comment (waldner) [reply piccolo/apps/user/commands/list.py:13]: Depends on how much of a purist you are. Personally, I'd just leave it with no output at all, but your call.

File: piccolo/apps/user/commands/list.py (hunk 1 above)
Comment (waldner) [reply piccolo/apps/user/commands/list.py:10]: Done.

File: piccolo/apps/user/commands/list.py (hunk 3 above)
Comment (waldner) [reply piccolo/apps/user/commands/list.py:26]: Done both things.

File: piccolo/apps/user/commands/list.py (hunk 1 above)
Comment (dantownsend) [reply piccolo/apps/user/commands/list.py]: You're right - not a big problem, lets leave it.

File: piccolo/apps/user/commands/list.py (hunk 2 above)
Comment (dantownsend) [reply piccolo/apps/user/commands/list.py]: My only concern with not printing something out is the user might think it's broken.

Comment (waldner) [reply piccolo/apps/user/commands/list.py]: Alright, I added a message.

**Comment by dantownsend** [comment]
OK, this looks good - thanks! I'll merge it in.

I'll add a really basic test for code coverage.

**Comment by waldner** [comment]
Thanks!

**Pull request merged**